src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {
+  memo, useCallback, useEffect, useState,
+} from 'react';
+import { useParams } from 'react-router-dom';
+import { Loader } from '../components/Loader';
+import { PersonTable } from '../components/PersonTable/PersonTable';
+import { Person } from '../types';
+
+export const PeoplePage: React.FC = memo(() => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const { slug } = useParams();
+  const [errorText, setErrorText] = useState('');
+
+  const errorMessage = (error: string) => {
+    setErrorText(error);
+  };
+
+  const peopleFromServer = useCallback(async () => {
+    try {
+      const response = await fetch(
+        'https://mate-academy.github.io/react_people-table/api/people.json',
+      );
+
+      const personsFromServer = await response.json();
+
+      const peopleWithParents = personsFromServer.map((person: Person) => {
+        return {
+          ...person,
+          mother: personsFromServer.find(
+            (per: Person) => per.name === person.motherName,
+          ),
+          father: personsFromServer.find(
+            (per: Person) => per.name === person.fatherName,
+          ),
+        };
+      });
+
+      setPeople(peopleWithParents);
+    } catch {
+      errorMessage('Something went wrong');
+    }
+  }, []);
+
+  useEffect(() => {
+    peopleFromServer();
+  }, []);
+  const tabbleHeaderColumns
+    = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
```

Comment:
Use `useMemo` or move it from the component.

Also, you could try `Object.keys(people[0])` and capitalize words